File: betCalc.js
// Get user input on the odds and how much is being bet (this is for use in Node JS):
const prompt = require('prompt-sync')();

let odds = prompt('What are the odds of your bet?: ');
odds = parseInt(odds);
let stake = prompt('How much do you intend to bet?: ');
stake = parseInt(stake);

// Function for determining the probability and payout:

const odds_payout = (odds, stake) => {
    if (odds > 0) {
        oddsMultiplier = odds/100;
        let prob = odds/(odds+100);
        let probRounded = Math.round((prob + Number.EPSILON) * 100)/100;
        let probPct = probRounded * 100;
        let profit = stake * oddsMultiplier;
        let payout = stake + profit;
        return `The probability of winning is ${probPct} percent and the potential payout is ${payout} dollars.`
    } else if (odds < 0) { 
        oddsMultiplier = Math.abs(odds/100);       
        let prob = Math.abs(odds)/(Math.abs(odds) + 100);
        let probRounded = Math.round((prob + Number.EPSILON) * 100)/100;
        let probPct = probRounded * 100;
        let profit = stake / oddsMultiplier;
        let payout = stake + profit;
        return `The probability of winning is ${probPct} percent and the potential payout is ${payout} dollars.`
    }
    
}

console.log(odds_payout(odds, stake));

File: bet_calc.py
odds = int(input('What are the odds of your bet?: '))
stake = int(input('How much do you intend to bet?: '))

# Function to calculate probability and potential payout

def calculator(odds, stake):

	if odds > 0:
		odds_multiplier = odds/100
		implied_prob = round(100/(odds + 100), 2)
		profit = round(stake * odds_multiplier,2)
		return f'The implied probability of winning the bet is {round(implied_prob * 100, 2)}% and your potential payout is: {round(profit + stake, 2)} dollars.'
	elif odds < 0:
		odds_multiplier = abs(odds/100)
		implied_prob = round(abs(odds)/(abs(odds) + 100), 2)
		profit = round(stake / odds_multiplier,2)
		return f'The implied probability of winning the bet is {round(implied_prob * 100, 2)}% and your potential payout is: {round(profit + stake, 2)} dollars.'

print(calculator(odds, stake))